teerex/src/lib.rs

```diff
@@ -40,72 +39,130 @@ use ias_verify::SgxBuildMode;
+		UnshieldedFunds(T::AccountId),
+		ProcessedParentchainBlock(T::AccountId, H256, H256),
+		ProposedSidechainBlock(T::AccountId, H256),
+	}
+
+	// Watch out: we start indexing with 1 instead of zero in order to
+	// avoid ambiguity between Null and 0.
+	#[pallet::storage]
+	#[pallet::getter(fn enclave)]
+	pub type EnclaveRegistry<T: Config> =
+		StorageMap<_, Blake2_128Concat, u64, Enclave<T::AccountId, Vec<u8>>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn enclave_count)]
+	pub type EnclaveCount<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn enclave_index)]
+	pub type EnclaveIndex<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, u64, ValueQuery>;
+
+	// Enclave index of the worker that recently committed an update.
+	#[pallet::storage]
+	#[pallet::getter(fn worker_for_shard)]
+	pub type WorkerForShard<T: Config> =
+		StorageMap<_, Blake2_128Concat, ShardIdentifier, u64, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn confirmed_calls)]
+	pub type ExecutedCalls<T: Config> = StorageMap<_, Blake2_128Concat, H256, u64, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn allow_sgx_debug_mode)]
+	pub type AllowSGXDebugMode<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub allow_sgx_debug_mode: bool,
 	}
-);
-
-decl_storage! {
-	trait Store for Module<T: Config> as Teerex {
-		// Simple lists are not supported in runtime modules as theoretically O(n)
-		// operations can be executed while only being charged O(1), see substrate
-		// Kitties tutorial Chapter 2, Tracking all Kitties.
-
-		// watch out: we start indexing with 1 instead of zero in order to
-		// avoid ambiguity between Null and 0
-		pub EnclaveRegistry get(fn enclave): map hasher(blake2_128_concat) u64 => Enclave<T::AccountId, Vec<u8>>;
-		pub EnclaveCount get(fn enclave_count): u64;
-		pub EnclaveIndex get(fn enclave_index): map hasher(blake2_128_concat) T::AccountId => u64;
-		// enclave index of the worker that recently committed an update
-		pub WorkerForShard get(fn worker_for_shard) : map hasher(blake2_128_concat) ShardIdentifier => u64;
-		pub ExecutedCalls get(fn confirmed_calls): map hasher(blake2_128_concat) H256 => u64;
-		pub AllowSGXDebugMode get(fn allow_sgx_debug_mode) config(allow_sgx_debug_mode): bool;
+
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self { allow_sgx_debug_mode: Default::default() }
+		}
 	}
-}
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		type Error = Error<T>;
-		fn deposit_event() = default;
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			{
+				let data = &self.allow_sgx_debug_mode;
+				let v: &bool = data;
+				<AllowSGXDebugMode<T> as frame_support::storage::StorageValue<bool>>::put::<&bool>(
+					v,
+				);
+			}
```

Comment:
This one is unnecessarily verbose due to the migration template, you can simplify this to a one liner:

```
<AllowSGXDebugMode<T>>::put(&self.allow_sgx_debug_mode);

```

Reply:
Good catch, updated accordingly